src/main/java/com/exadel/movie_new/service/MovieServiceImpl.java

```diff
@@ -48,64 +49,37 @@ public List<Movie> getAllMovies() {
 
     @Override
     public Movie getMovie(String movieId) {
-        List<Movie> movieData = readJsonData();
+        List<Movie> movieData = movieDao.read();
         if (searchMovie(movieData, movieId).isEmpty()) {

```

Comment:
every time you call **searchMovie** you run stream. It's big mistake, because you are doing redundant operation. Please save result of the first **searchMovie** into variable, and use it in the next step.